### PR TITLE
Various small fixes: text selection in tables, cache issues

### PR DESCRIPTION
**Pull request by poire-z**
See individual commit messages for details.

About _Add known HTML attributes to avoid cache issues_ : 
I had some "style hash mismatch, doing full rendering" once I applied the `In-page EPUB footnotes` style tweak, because it contains:
```
*[type~="rearnote"],
*[role~="doc-note"],
```
The issue disappeared when I added `XS_ATTR( role )` in fb2def.h.
Unspecified attributes are given an incremented ID, which may vary depending on the order of some operations. Specified attributes in this files get a fixed ID. Couldn't pinpoint the real issue (there's some sorting of names/ids when serializing/deserializing, might be that).
So, to avoid further issues, I decided to add a huge amount of known HTML attributes, in case some publisher uses them in their own CSS. That list just makes out a few #define, and may add a few 100bytes to each cache file.

About _Make existing context.AddLine() and pb_flag clearer_ 
That's just to make things easier to read. And future diff easier to read if I add some `pb_flag |=` around these lines. Rather have that early to be sure I didn't mess these changes :)

### CODE REVIEW

File: crengine/include/fb2def.h [hunk 1]
```diff
@@ -217,6 +220,227 @@ XS_ATTR( title )
...
+XS_ATTR( kind )
+XS_ATTR( label )
+XS_ATTR( language )
+XS_ATTR( lazyload )
+XS_ATTR( list )
+XS_ATTR( longdesc )
+XS_ATTR( loop )
+XS_ATTR( low )
+XS_ATTR( manifest )
+XS_ATTR( marginheight )
```
Comment (Frenzie) [inline crengine/include/fb2def.h]: Might as well be complete but I'd like to see that EPUB with frames… :rofl:

**Comment by poire-z** [comment]
Is that codacy stuff stuck or something? No progress log that I can see :|

File: crengine/include/fb2def.h (hunk 1 above)
Comment (poire-z) [reply crengine/include/fb2def.h]: I still have a few 10+ years old HTML files saved (from the days we used web crawlers to grab a web site - haven't done that for quite some time :)

**Comment by Frenzie** [comment]
Weird, maybe it's in a queue or something? But since Travis also runs cppcheck there's nothing it should have to add.

clang-tidy does seem to have a few things to say about this file but not about the changes.

**Comment by Frenzie** [comment]
Oh wait, codacy also runs flawfinder: https://dwheeler.com/flawfinder/

https://app.codacy.com/app/Frenzie/crengine_2/issues/index?bid=11379228

**Pull request merged**